firefox_puppeteer/tests/test_windows.py

```diff
@@ -3,98 +3,181 @@
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 from marionette import By
-from marionette.errors import NoSuchElementException
 
+import firefox_puppeteer.errors as errors
 from firefox_ui_harness.testcase import FirefoxTestCase
 
+from firefox_puppeteer.ui.windows import BaseWindow
+
 
 class TestWindows(FirefoxTestCase):
 
-    def test_switch_to_window(self):
+    def test_windows(self):
         url = self.marionette.absolute_url('layout/mozilla')
 
-        first_window = self.windows.current
-
-        # Those actions will fail if the current window is not a browser window
-        first_window.send_shortcut('ctrl-n')
-        first_window.send_shortcut('ctrl-n')
+        # Open two more windows
+        for index in range(0, 2):
+            self.marionette.execute_script(""" window.open(); """)
 
         windows = self.windows.all
-        self.assertEquals(len(windows), 3)
+        self.assertEquals(len(windows), len(self.marionette.chrome_window_handles))
 
+        # Switch to the 2nd window
+        self.windows.switch_to(windows[1].handle)
+        self.assertEquals(windows[1].handle, self.marionette.current_chrome_window_handle)
+
+        # Make the window unique by its loaded URL
```

Comment:
The comment doesn't match any code here. So its a left-over from the very first experiments I think.